crates/api_common/src/send_activity.rs

```diff
@@ -105,6 +106,18 @@ pub enum SendActivityData {
     report_creator: Person,
     community: Community,
   },
+  CreateReportToSite {
+    object_id: Url,
+    actor: Person,
+    site: Site,
+    reason: String,
+  },
+  SendResolveReportToSite {
+    object_id: Url,
+    actor: Person,
+    report_creator: Person,
+    site: Site,
```

Comment:
Should be called `CreateCommunityReport` and `ResolveCommunityReport` to be clear.

Reply:
On L97, why not just split out all the different report types like we do for everything else here?

IE `CreateCommentReport`, `CreatePrivateMessageReport`, etc

Reply:
Its unnecessary because they are federated with the same activity, so it would only lead to duplicate code which does the exact same thing.

Also the name is probably fine as is because we might also want to federate private message reports later. And `CreateCommunityReport` can sound like reporting something *in* a community.

Reply:
I removed these variants and made the other two report variants use `Either`